Sources/SpeziOpenAI/MessagesView.swift

```diff
@@ -67,32 +66,31 @@ public struct MessagesView: View {
     /// - Parameters:
     ///   - chat: The chat messages that should be displayed.
     ///   - bottomPadding: A fixed bottom padding for the messages view.
-    ///   - hideSystemMessages: If system messages should be hidden from the chat overview.
+    ///   - hideMessagesWithRoles: The .system and .function roles are hidden from message view
     public init(
         _ chat: [Chat],
-        bottomPadding: CGFloat = 0,
-        hideSystemMessages: Bool = true
+        hideMessagesWithRoles: Set<Chat.Role> = [.system, .function],
+        bottomPadding: CGFloat = 0
     ) {
         self._chat = .constant(chat)
+        self.hideMessagesWithRoles = hideMessagesWithRoles
         self._bottomPadding = .constant(bottomPadding)
-        self.hideSystemMessages = hideSystemMessages
     }
-    
-    
+
     /// - Parameters:
     ///   - chat: The chat messages that should be displayed.
     ///   - bottomPadding: A bottom padding for the messages view.
-    ///   - hideSystemMessages: If system messages should be hidden from the chat overview.
+    ///   - hideMessagesWithRoles: Defines which messages should be hidden based on the passed in message roles.
     public init(
         _ chat: Binding<[Chat]>,
-        bottomPadding: Binding<CGFloat> = .constant(0),
-        hideSystemMessages: Bool = true
+        hideMessagesWithRoles: Set<Chat.Role> = [.system, .function],
```

Comment:
As discussed in the previous review, you can also use the default parameter here:
```suggestion
        hideMessagesWithRoles: Set<Chat.Role> = MessageView.Defaults.hideMessagesWithRoles,
```